### PR TITLE
Add script to move firedrake plus license.

### PR DESCRIPTION
**Pull request by nickaj**
This adds a script which allows one to move an intact firedrake install around their filesystem
without breaking it. Also updates the license (as it is a derivative work) and readme.

### CODE REVIEW

File: scripts/clone.py [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+if sys.platform == 'win32':
+    env_bin_dir = 'Scripts'
+
+class UserError(Exception):
+    pass
+
+def in_directory(path, tree):
+    "Make both paths absolute."
+    directory = os.path.join(os.path.realpath(tree), '')
+    file = os.path.realpath(path)
```
Comment (wence-) [inline scripts/clone.py]: unused variable?

File: scripts/clone.py [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+    >>> _dirmatch('/home/foo/bar2', '/home/foo/bar')
+    False
+    >>> _dirmatch('/home/foo/bar2/etc', '/home/foo/bar')
+    False
+    """
+    matchlen = len(matchwith)
+    if (path.startswith(matchwith)
+        and path[matchlen:matchlen + 1] in [os.sep, '']):
+        return True
+    return False
```
Comment (wence-) [inline scripts/clone.py]: AKA `return condition`

File: scripts/clone.py [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+    logger.debug("stdout from sed: %s" % stdout)
+    logger.debug("errcode from sed: %s" % errcode)
+    
+
+def clone_virtualenv(src_dir, dst_dir, do_copy=True):
+    """ Central Function: Move the whole directory tree, except .pyc files """
+    
+    # It is possible that you want to go through the process more than once, but the second time, you don't want to do the copy.
+    # This is risky, as if do_copy is False, you will skip the checks on file existence etc.
+    if do_copy==True:
```
Comment (wence-) [inline scripts/clone.py]: `if do_copy`

**Comment by wence-** [comment]
I haven't read through this properly.  What I would like is that this is *absolutely* failsafe.  That is, if, for whatever reason, the script falls over halfway through, does it leave my virtualenv in an unusable state?  That would be bad.  I presume this is a one-off thing: I create my virtualenv somehow, then I run this script (how?).  Once I've done that, I'm ready to use the new "boxed" virtualenv.  So presumably my original virtualenv is always "read-only".

**Comment by wence-** [comment]
Is the current status that we are going to do this as a stand-alone script (not in the firedrake repo?).

**Comment by nickaj** [comment]
Yes, please could you drop this PR and generate a new repo under firedrakeproject that I can issue PRs against (so someone from the team reviews the code).

**Comment by wence-** [comment]
Done, see https://github.com/firedrakeproject/relocate-venv

**Pull request closed without merging**